reana-ui/src/pages/status/Status.js

```diff
@@ -0,0 +1,135 @@
+import { Container, Grid, Icon, Label, Loader } from "semantic-ui-react";
+import axios from "axios";
+
+import BasePage from "../BasePage";
+import { errorActionCreator } from "~/actions";
+import { Title, PieChart } from "~/components";
+import { api } from "~/config";
+import { healthMapping } from "~/util";
+
+import styles from "./Status.module.scss";
+
+export default function Status() {
+  const [status, setStatus] = useState();
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const getClusterStatus = () => {
+      setLoading(true);
+      axios({
+        method: "get",
+        url: api + "/api/status",
+        withCredentials: true,
+      })
+        .then((res) => {
+          setStatus(res.data);
+          setLoading(false);
+        })
+        .catch((err) => {
+          setStatus({});
+          setLoading(false);
+          dispatch(errorActionCreator(err));
+        });
+    };
+
+    getClusterStatus();
+  }, []);
+
+  const serialize = {
+    node: ({ available, unschedulable, ...rest }) => ({
+      title: "Nodes",
+      details: [`${available} available`, `${unschedulable} unschedulable`],
+      ...rest,
+    }),
+    workflow: ({ running, queued, ...rest }) => ({
+      title: "Workflows",
+      details: [`${running} running`, `${queued} queued`],
+      ...rest,
+    }),
+    job: ({ running, pending, ...rest }) => ({
+      title: "Jobs",
+      details: [`${running} running`, `${pending} pending`],
+      ...rest,
+    }),
+    session: ({ active, ...rest }) => ({
+      title: "Notebooks",
+      details: [`${active} active`],
+      ...rest,
+    }),
+  };
+
+  const pieChartHealthMapping = {
+    healthy: "#dbbf2b", // yellow
+    warning: "#e5975e", // orange
+    critical: "#e55e5e", // red
+  };
+
+  const renderPieChart = ({ title, details, percentage, health }) => {
+    return (
+      <Grid.Column className={styles.column} key={title}>
+        <PieChart
+          value={percentage}
+          totalValue={100}
+          fillColor={pieChartHealthMapping[health]}
+          backgroundColor="mediumseagreen"
+        />
+        <div className={styles["status-details"]}>
+          <div className={styles.usage}>
+            <h3>{title}</h3>
+            {details.map((detail, index) => (
+              <div key={`${title}-${index}`}>{detail}</div>
+            ))}
+          </div>
+
+          {percentage !== undefined && (
+            <Label
+              basic
+              size="small"
+              color={healthMapping[health]}
+              className={styles.percentage}
+            >
+              {percentage || 0}%
+            </Label>
+          )}
+        </div>
+      </Grid.Column>
+    );
+  };
+
+  return (
+    <BasePage>
+      <Container text className={styles.container}>
+        <Title>
+          Cluster Health
+          <Icon name="heartbeat" />
+        </Title>
+        {loading || !status ? (
```

Comment:
Minor: right now if something goes wrong with the request (500 for example), we still show that the cluster is loading. We could provide something went wrong message for the user, WDYT?

Reply:
ah, I see you now dispatch `errorActionCreator` on error, nice. I got a bit outdated version locally..